miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/display/MiniAppWebViewClient.kt

```diff
@@ -14,10 +14,24 @@ internal class MiniAppWebViewClient(
     private val customScheme: String
 ) : WebViewClient() {
 
-    override fun shouldInterceptRequest(
-        view: WebView,
+    override fun shouldInterceptRequest(view: WebView, request: WebResourceRequest): WebResourceResponse? {
+        val interceptedWebRequest = loader.shouldInterceptRequest(request.url)
+
+        interceptMimeType(interceptedWebRequest, request)
```

Comment:
From SRP point of view, it makes sense to separate the functionality into a new method. 

Though, because I think this callback is quite related to the rendering cycle of the mini-app in WebView, I am not sure if applying code within the same method has faster response in millis. @minh-rakuten do you think we can check which is faster without a jump to new method in call stack or without jumping?

Also, because, this is invoked for all the resources, the `if` would always be executing. I was looking at issue [issue](https://bugs.chromium.org/p/chromium/issues/detail?id=824651). Should we really support it or provide documentation on an alternate solution? What do you think @corycaywood @minh-rakuten?

Reply:
I will put it in the same block scope then.

I don't have an idea of a decision. All I can tell is This is a bug of android webview itself and had been fixed since Android 10.

Reply:
As I mentioned above, I think we do need to support this because the `module` type provides some benefits to mini app developers. It doesn't seem like this should have a significant performance impact as long as the Mini App has a reasonable number of files.

By the way @ts-leojoseph-r @Climbatize Are there any issues with using `type=module` on iOS side?

Reply:
iOS had the problem with Rakuten Casa `type=module` too.
But iOS had fixed it in this [PR](https://github.com/rakutentech/ios-miniapp/pull/96).

Solution for both android & ios is defining MIME type.

Reply:
Yeah we had to artificially provide mime types in order to make Case work on iOS

Reply:
From the look of it, I understand that we might have to support it as a mimetype even though it seems to be a limitation on the web tech of not being able to provide the correct mimetype when this type - module is selected.

At the same time, number of files inside a mini-app would be out of control area.

@minh-rakuten, could you share the time diff from the extra hop? 
UT can take a back seat on this.

Reply:
@ts-nitish-a-jain 
Loading Rakuten Casa

Separate method: 43 ms
Same scope: 35 ms

Reply:
Couldn't you make it an `inline` function?

Reply:
I doubt the efficient of using `inline` here 😂.

Reply:
To clear the doubt, once you profile that section and scale it up with an assumption of 1000/10000 files, you will probably find it out.